include/mata/nfa.hh

```diff
@@ -992,31 +992,15 @@ Nfa construct(
         StringToSymbolMap*                   symbol_map = nullptr,
         StringToStateMap*                    state_map = nullptr)
 { // {{{
-    Nfa aut;
-
-    bool remove_symbol_map = false;
-    if (nullptr == symbol_map)
-    {
-        symbol_map = new StringToSymbolMap();
-        remove_symbol_map = true;
+    StringToSymbolMap tmp_symbol_map;
```

Comment:
Why do you need this temp thing here?

Reply:
It's not needed, but it's safer than using new/delete.